src/forms/ProfileSettingsForm/ProfileSettingsForm.js

```diff
@@ -25,10 +25,10 @@ class ProfileSettingsFormComponent extends Component {
     this.submittedValues = {};
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
     // Upload delay is additional time window where Avatar is added to the DOM,
     // but not yet visible (time to load image URL from srcset)
-    if (this.props.uploadInProgress && !nextProps.uploadInProgress) {
+    if (prevProps.uploadInProgress && !this.props.uploadInProgress) {
       this.setState({ uploadDelay: true });
       this.uploadDelayTimeoutId = window.setTimeout(() => {
```

Comment:
There's a bug in here: componentWillUnmount function should refer to this timeout.
```
  componentWillUnmount() {
    window.clearTimeout(this. uploadDelayTimeoutId);
  }
```

Reply:
Although, it is probably better to fix this in a separate PR.

Reply:
https://github.com/sharetribe/flex-template-web/pull/1185